src/utils/image-processor.js

```diff
@@ -1,37 +1,42 @@
 import $ from 'jquery'
 
 import { Blob, FileReader, URL } from '../utils/abilities'
-import { imageLoader } from '../utils/image-loader'
+import { imageLoader } from '../utils/image-loader.ts'
 import { defer, fitSize, canvasToBlob, taskRunner } from '../utils'
 import { isWindowDefined } from './is-window-defined'
 
 // utils image
-var DataView = isWindowDefined() && window.DataView
-var runner = taskRunner(1)
+const DataView = isWindowDefined() && window.DataView
+const runner = taskRunner(1)
 
 const shrinkFile = function(file, settings) {
+  // let res = () => {}
+  let rej = () => {}
+  const promise = new Promise((resolve, reject) => {
+    // res = resolve
+    rej = reject
+  })
   var df
   // in -> file
   // out <- blob
   df = $.Deferred()
```

Comment:
if you didn't remove $ from this function, maybe will be better return it original code?
Or finish

Reply:
Yes, you are right. Rollback this